src/software/ai/hl/stp/tactic/move_tactic.h

```diff
@@ -1,7 +1,55 @@
 #pragma once
 
-#include "software/ai/hl/stp/action/move_action.h"
+#include "software/ai/hl/stp/action/move_action.h"  // TODO (#1888): remove this dependency
 #include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+
+struct MoveFSM
+{
+    struct Move
+    { /* state */
+    };
+
+    struct Controls
+    {
+        // The point the robot is trying to move to
+        Point destination;
+        // The orientation the robot should have when it arrives at its destination
+        Angle final_orientation;
+        // The speed the robot should have when it arrives at its destination
+        double final_speed;
+    };
+
+    struct Update
+    {
+        Controls controls;
+        TacticUpdate common;
+    };
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto update_move_intent = [](auto event) {
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), event.controls.destination,
+                event.controls.final_orientation, event.controls.final_speed,
+                DribblerMode::OFF, BallCollisionType::AVOID));
+        };
+
+        const auto movement_done = [](auto event) {
+            return moveRobotDone(event.common.robot, event.controls.destination,
+                                 event.controls.final_orientation);
+        };
+
+        return make_transition_table(
+            *"idle"_s + event<Update> / update_move_intent = state<Move>,
```

Comment:
This table will look at confusing to new members. If this is used as the reference for FSM tactics, it would be good to label the columns of the table with src state, event, guard, and dest state.

Reply:
done

Reply:
Do you do this in every tactic though? One option that I've used as a middle ground on a few student teams is to have 1 instance of each class that's really heavily commented so that you can direct new people to look at that as a reference without having to duplicate comments.